Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6320,6 +6320,8 @@ void WebPageProxy::didCommitLoadForFrame(FrameIdentifier frameID, FrameInfoData&
     if (m_userMediaPermissionRequestManager)
         m_userMediaPermissionRequestManager->didCommitLoadForFrame(frameID);
 #endif
+
+    platformDidCommitLoadForFrame(*frame);
```

Comment:
The logic in this method could have just been inlined in the `if (frame->isMainFrame()) {` on line 6297, like the other features?

Reply:
Yep, I can make this change! I only did it this way to avoid importing the ObjC header `WKTextExtractionUtilities.h` header in `WebPageProxy.cpp`, but I can instead replace the `@class` with the C++-safe `OBJC_CLASS` macro, and it should work fine.